src/AggregateRoots/AggregateRoot.php

```diff
@@ -116,6 +125,25 @@ public function recordThat(ShouldBeStored $domainEvent): static
         return $this;
     }
 
+    public function recordConcurrently(ShouldBeStored $domainEvent, bool|callable $allowConcurrent = true): static
```

Comment:
What's the use case for being able to pass a boolean to `$allowConcurrent`.

It's kind of confusing to me that you'd call `recordConcurrently`, and then pass `false` to force it not to be concurrent.

Reply:
Ideally the signature would be `true|callable` to allow concurrency without a check, but that doesn't exist in PHP 🙃 So instead I supported booleans in general, which means we have to handle false too (even though it'd never be used)

I agree this is confusing, I updated the signature to `?callable`, when `$allowConcurrent` is `null` we won't do any extra checks.

Reply:
FYI `true` is a standalone type in 8.2: https://stitcher.io/blog/new-in-php-82#null,--true,-and-false-as-standalone-types-rfc

Reply:
Good to know! Still supporting 8.1 for now :)